src/viam/sdk/common/version_metadata.cpp

```diff
@@ -0,0 +1,41 @@
+#include <viam/sdk/common/version_metadata.hpp>
+
+#include <sstream>
+
+#include <viam/sdk/common/private/version_metadata.hpp>
+
+namespace viam {
+namespace sdk {
+
+std::string sdk_version() {
+    std::string version_metadata(impl::k_version);
+    version_metadata.erase(0, version_metadata.find(';') + 1);
+    return version_metadata.substr(0, version_metadata.find(';'));
+}
+
+int get_sub_version(int which) {
+    auto version = sdk_version();
```

Comment:
Similarly here, we could do something like

```cpp
static const std::array<int, 3> components =  [] {
    std::array<int, 3> result;
    
    // copy your function here but let the `for` loop parse the `int`s into the array

    return result;
}();
```

and then your function just returns `components[i]`;